immer/config.hpp

```diff
@@ -18,6 +18,28 @@
 #endif
 #endif
 
+#ifdef __has_feature
+#if !__has_feature(cxx_exceptions)
```

Comment:
I tried having these two checks in a single line to directly choose the version of the macros to use, but that caused build errors like this:
```
/usr/local/google/home/carlosk/lib/immer/immer/config.hpp:21:45: error: missing binary operator before token "("
   21 | #if !defined(__has_feature) || __has_feature(cxx_exceptions)
      |    
```
Splitting into two separate lines like this worked.